src/Servers/Kestrel/Kestrel/test/HttpsConfigurationTests.cs

```diff
@@ -21,35 +23,39 @@ public class HttpsConfigurationTests
     [InlineData("https://127.0.0.1:0", false)]
     public async Task BindAddressFromSetting(string address, bool useKestrelHttpsConfiguration)
     {
-        var hostBuilder = new WebHostBuilder()
-                .UseKestrelCore()
-                .ConfigureKestrel(serverOptions =>
-                {
-                    serverOptions.TestOverrideDefaultCertificate = new X509Certificate2(Path.Combine("shared", "TestCertificates", "aspnetdevcert.pfx"), "testPassword");
-                })
-                .Configure(app => { });
-
-        // This is what ASPNETCORE_URLS would populate
-        hostBuilder.UseSetting(WebHostDefaults.ServerUrlsKey, address);
-
-        if (useKestrelHttpsConfiguration)
-        {
-            hostBuilder.UseKestrelHttpsConfiguration();
-        }
+        var hostBuilder = new HostBuilder()
+            .ConfigureWebHost(webHostBuilder =>
+            {
+                webHostBuilder
+                    .UseKestrelCore()
+                    .ConfigureKestrel(serverOptions =>
+                    {
+                        serverOptions.TestOverrideDefaultCertificate = new X509Certificate2(Path.Combine("shared", "TestCertificates", "aspnetdevcert.pfx"), "testPassword");
+                    })
+                    .Configure(app => { })
+                    // This is what ASPNETCORE_URLS would populate
+                    .UseSetting(WebHostDefaults.ServerUrlsKey, address);
 
-        var host = hostBuilder.Build();
+                if (useKestrelHttpsConfiguration)
+                {
+                    webHostBuilder.UseKestrelHttpsConfiguration();
+                }
+            });
 
-        Assert.Single(host.ServerFeatures.Get<IServerAddressesFeature>().Addresses, address);
+        using var host = hostBuilder.Build();
 
         if (address.StartsWith("https", StringComparison.OrdinalIgnoreCase) && !useKestrelHttpsConfiguration)
         {
-            Assert.Throws<InvalidOperationException>(host.Run);
+            Assert.Throws<InvalidOperationException>(host.Start);
+            Assert.Empty(host.Services.GetRequiredService<IServer>().Features.Get<IServerAddressesFeature>().Addresses);
         }
         else
         {
-            // Binding succeeds
             await host.StartAsync();
-            await host.StopAsync();
+
+            var addr = Assert.Single(host.Services.GetRequiredService<IServer>().Features.Get<IServerAddressesFeature>().Addresses);
```

Comment:
Addresses aren't set until the host starts, and the address(es) contain the allocated port. Just want to call it our since it's different behavior from WebHostBuilder.